crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -1422,6 +1422,13 @@ impl PaymentCreate {
             .attach_printable("Unable to encrypt the payment intent data")?;
 
         let skip_external_tax_calculation = request.skip_external_tax_calculation;
+        let tax_amount = request.order_tax_amount.unwrap_or(MinorUnit::new(0));
+        let tax_details = Some(diesel_models::TaxDetails {
+            default: Some(diesel_models::DefaultTax {
+                order_tax_amount: tax_amount,
+            }),
+            ..Default::default()
```

Comment:
Can we explicitly populate all the fields instead of using `..Default::default()`? This would enforce us to also populate values for any new fields that may be introduced on a later date.